build_tools/cmake/iree_lit_test.cmake

```diff
@@ -0,0 +1,94 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+include(CMakeParseArguments)
+
+# iree_lit_test()
+#
+# Creates a lit test for the specified source file.
+#
+# Mirrors the bzl rule of the same name.
+#
+# Parameters:
+# NAME: Name of the target
+# TEST_FILE: Test file to run with the lit runner.
+# DATA: Additional data dependencies invoked by the test (e.g. binaries
+#   called in the RUN line)
+#
+# TODO(gcmn): allow using alternative driver
+# A driver other than the default iree/tools/run_lit.sh is not currently supported.
+function(iree_lit_test)
+  cmake_parse_arguments(
+    _RULE
+    ""
+    "NAME;TEST_FILE"
+    "DATA"
+    ${ARGN}
+  )
+  if(NOT IREE_BUILD_TESTS)
+    return()
+  endif()
+
+  iree_package_name(_PACKAGE_NAME)
+  set(_NAME "${_PACKAGE_NAME}_${_RULE_NAME}")
+
+  get_filename_component(_TEST_FILE_PATH ${_RULE_TEST_FILE} ABSOLUTE)
+  
+  add_test(
+    NAME ${_NAME}
+    COMMAND ${CMAKE_SOURCE_DIR}/iree/tools/run_lit.sh "${_TEST_FILE_PATH}"
+    WORKING_DIRECTORY "${CMAKE_BINARY_DIR}" # Make sure the lit runner can find all the binaries
+  )
+  set_tests_properties(${_NAME} PROPERTIES DEPENDS _RULE_DATA)
```

Comment:
I'm actually not sure this is necessary. The [documentation for DEPENDS](https://cmake.org/cmake/help/latest/prop_test/DEPENDS.html) says this means it should be run after other tests, but these are binaries and should be created at build time.

Reply:
Seems the deps are not needed. Might have been a different (unrelated) issue, that caused FileCheck not be build on my local machine. Will dig into this.

Reply:
K I will remove them for now. We do want to say that building this target is dependent on building those others though. Not sure how to best do that.